### PR TITLE
Refactor for complexity (and some cleanup in the process)

### PR DESCRIPTION
**Pull request by CoMPaTech**
Just a PR so codefactor can keep up (skips checking radon on cli and/or sifting code through copy/paste on lizard.ws)

Main focus was reducing complexity before we completely refactor the setup. (And not interfering with the `codefactor` branch/PR)

### CODE REVIEW

File: plugwise/helper.py [hunk 1]
```diff
@@ -698,6 +710,23 @@ def heating_valves(self):
 
         return None if loc_found == 0 else open_valve_count
 
+    @staticmethod
+    def energy_diff(measurement, net_string, f_val, direct_data):
```
Comment (bouwew) [inline plugwise/helper.py:714]: If you move this one outside of the class, to the top of the file, there's no need to use `@static_method` and you can call it by using the function-name, no longer needed to call it via `self`.

File: plugwise/helper.py [hunk 2]
```diff
@@ -276,109 +341,56 @@ def all_appliances(self):
...
-            appliance_types = set()
-
-            appliance_id = appliance.attrib["id"]
-            appliance_class = appliance.find("type").text
-            appliance_name = appliance.find("name").text
-            appliance_model = appliance_class.replace("_", " ").title()
-            appliance_fw = None
-            appliance_v_name = None
-
+            appl = Munch()
```
Comment (bouwew) [inline plugwise/helper.py:344]: Nice! First time I hear about Munch() :)

File: plugwise/helper.py (hunk 1 above)
Comment (CoMPaTech) [reply plugwise/helper.py:714]: if we would use it across (multiple) classes (from that same include) that would make sense, so if we envision the same being used by USB that indeed would be something to move

File: plugwise/helper.py (hunk 2 above)
Comment (CoMPaTech) [reply plugwise/helper.py:344]: Me too :) But it makes for much more object-oriented approach and less fugly dict-approach (and I really like it so far)

Comment (bouwew) [reply plugwise/helper.py:344]: :+1:

File: plugwise/helper.py [hunk 3]
```diff
@@ -239,11 +243,98 @@ def all_locations(self):
 
         return
 
+    def all_appliances_class_finder(self, appliance, appl):
```
Comment (bouwew) [inline plugwise/helper.py:246]: Sorry, not a good name, I would propose `appliance_info_finder` or something similar as it finds: firmware version, model, name and vendor name, to be used as data for the Core `device_info`.

File: plugwise/helper.py [hunk 4]
```diff
@@ -239,11 +243,98 @@ def all_locations(self):
...
+            module_data = self.get_module_data(appliance, locator, mod_type)
+            appl.v_name = module_data[0]
+            appl.model = version_to_model(module_data[1])
+            appl.fw = module_data[3]
+            return appl
+
+        # Cornercase just return existing dict-object
+        return appl  # pragma: no cover
+
+    def all_appliances_types_finder(self, appliance, appl):
```
Comment (bouwew) [inline plugwise/helper.py:311]: I would suggest to remove `all_` from the function name.

File: plugwise/helper.py (hunk 3 above)
Comment (CoMPaTech) [reply plugwise/helper.py:246]: reason behind the maddness: `all_appliances` function now has `all_appliances_CLASS...` and `all_appliances_TYPE...`

File: plugwise/helper.py (hunk 4 above)
Comment (CoMPaTech) [reply plugwise/helper.py:311]: see above, the reasoning being them partials - but i'm all up for shortening the names

File: plugwise/smile.py [hunk 5]
```diff
@@ -214,28 +228,8 @@ def get_all_devices(self):
...
-                if outdoor_temperature is not None:
-                    device_data["outdoor_temperature"] = outdoor_temperature
-
-            # Try to get P1 data and 2nd outdoor_temperature, when present
-            power_data = self.power_data_from_location(details["location"])
-            if power_data is not None:
-                device_data.update(power_data)
-
-        # Switching Groups
+    def get_device_data_switching_groups(self, details, device_data):
```
Comment (bouwew) [inline plugwise/smile.py:231]: Suggest to remove `get_` from the names of 4 device_data sub functions.

Comment (CoMPaTech) [reply plugwise/smile.py]: 👍

File: plugwise/helper.py (hunk 1 above)
Comment (bouwew) [reply plugwise/helper.py]: I was just being practical, I've moved them outside the class so they can be moved into `util.py` at a later moment, just like the other functions that are already in that file :)

Comment (CoMPaTech) [reply plugwise/helper.py]: I'll move it out next commit

**Review by CoMPaTech — COMMENTED** [review]
👍

**Pull request merged**